pinot-core/src/main/java/com/linkedin/pinot/core/startree/OffHeapStarTreeBuilder.java

```diff
@@ -645,9 +645,12 @@ private void constructStarTree(TreeNode node, int startDocId, int endDocId, int
     if (tempBufferSize > MMAP_SIZE_THRESHOLD) {
       // Create a temporary file and use MMapBuffer
       File tempFile = new File(_tempDir, startDocId + "_" + endDocId + ".unique.tmp");
-      try (FileChannel src = new FileInputStream(_dataFile).getChannel();
-          FileChannel dest = new FileOutputStream(tempFile).getChannel()) {
-        dest.transferFrom(src, startDocId * _docSize, tempBufferSize);
+      try (FileChannel src = new RandomAccessFile(_dataFile, "r").getChannel();
+          FileChannel dest = new RandomAccessFile(tempFile, "rw").getChannel()) {
+        long numBytesTransferred = src.transferTo(startDocId * _docSize, tempBufferSize, dest);
+        Preconditions.checkState(numBytesTransferred == tempBufferSize,
+            "Error transferring data from data file to temp file, transfer size mis-match");
+        dest.force(false);
```

Comment:
what is the idea behind calling force and especially with false?

Reply:
Ensure the data is flushed to disk (false means we don't enforce metadata to be flushed).

Reply:
this is in try catch block, which means close would be called at the end of it and that would internally call flush right?. btw, this if for my own understanding. I know having this additional line is probably harmless.

Reply:
@kishoreg Based on the javadoc, seems there is no guarantee on that.